tests/dream/io_test.py

```diff
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+import pytest
+
+from ess import dream
+
+
+@pytest.fixture
+def filename():
+    return dream.data.get_path('DREAM_nexus_sorted-2023-12-07.nxs')
+
+
+@pytest.mark.filterwarnings("ignore:Failed to load /entry/instrument/monitor_bunker")
+@pytest.mark.filterwarnings("ignore:Failed to load /entry/instrument/monitor_cave")
+@pytest.mark.filterwarnings("ignore:Failed to load /entry/instrument/polarizer/rate")
+@pytest.mark.filterwarnings("ignore:Failed to load /entry/instrument/sans_detector")
+def test_load_nexus_loads_file(filename):
+    dg = dream.load_nexus(filename)
+    assert 'instrument' in dg
+    instr = dg['instrument']
+    for name in (
+        'mantle',
+        'endcap_backward',
+        'endcap_forward',
+        'high_resolution',
+        'sans',
+    ):
+        assert f'{name}_detector' in instr
```

Comment:
In the GEANT4 loader, those are called `name`, i.e., without `"_detector"`. We should use the same names in both and I'm in favour of the shorter names in the GEANT4 loader. What do you think?